R/seroprevalence_data.R

```diff
@@ -120,4 +120,131 @@ prepare_bin_data <- function(serodata) {
     p_obs_bin_u = .data$Upper
   )
   return(xx)
+}
+
+# TODO: Complete the documentation of get_sim_counts
+#' Function that randomly generates a sample of counts for a simulated dataset
+#'
+#' @param sim_data A dataframe object containing the following columns:
+#' \tabular{ll}{
+#' \code{birth_year} \tab List of years in which the subjects were borned \cr \tab \cr
+#' \code{tsur} \tab Year of the survey\cr \tab \cr
+#' \code{country} \tab Default to 'none'.\cr \tab \cr
+#' \code{survey} \tab Survey label \cr \tab \cr
+#' \code{age_mean_f} \tab Age \cr \tab \cr
+#' }
+#' @return A simulated list of counts following a binomial distribution in accordance with a given force of infection and age class sizes.
+#' @examples
+#'\dontrun{
+#'
+#' }
+#' @export
+get_sim_counts <- function(sim_data, foi, size_age_class, seed = 1234){
+  exposure_ages <- get_exposure_ages(sim_data)
+  exposure_matrix <- get_exposure_matrix(sim_data)
+
+  set.seed(seed = seed)
+  sim_probabilities <- purrr::map_dbl(exposure_ages, ~1-exp(-pracma::dot(exposure_matrix[., ], foi)))
+  sim_counts <- purrr::map_int(sim_probabilities, ~rbinom(1, size_age_class, .))
+
+  return(sim_counts)
+}
+
+# TODO: Complete the documentation of generate_sim_data
+#' Function that generates simulated data from a given Force-of-Infection
+#'
+#' @param foi Numeric atomic vector corresponding to the desired Force-of-Infection
+#' @return Dataframe object containing the simulated data generated from \code{foi}
+#' @examples
+#'\dontrun{
+#' size_age_class = 5
+#' foi <- rep(0.02, 50)
+#' sim_data <- generate_sim_data(foi = foi,
+#'                               size_age_class = size_age_class,
+#'                               tsur = 2050,
+#'                               birth_year_min = 2000,
+#'                               survey_label = 'sim_constant_foi')
+#' }
+#' @export
+generate_sim_data <- function(foi,
+                              size_age_class,
+                              tsur,
```

Comment:
I'm not sure why a user needs to input `tsur` to simulate data?

Reply:
You're right, this information is not necessary to simulate the data. However, the function `extract_seromodel_summary` uses it to generate the models summary used for visualization purposes.  I'll restructure this in a future update in order to avoid conflicts with other branches we're working on right now; the package is a little unstable right now.

Reply:
This will be addressed by #100